src/main/java/team/chisel/ctm/client/util/VertexData.java

```diff
@@ -0,0 +1,148 @@
+package team.chisel.ctm.client.util;
+
+import com.mojang.blaze3d.vertex.VertexConsumer;
+import com.mojang.blaze3d.vertex.VertexFormatElement;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Getter;
+import lombok.ToString;
+import net.minecraft.world.phys.Vec2;
+import net.minecraft.world.phys.Vec3;
+
+@Getter
+@ToString
+public class VertexData {
+
+    private double posX, posY, posZ;
+    private float normalX, normalY, normalZ;
+
+    //Store int representations of the colors so that we don't go between ints and doubles when unpacking and repacking a vertex
+    private int red, green, blue, alpha;
+
+    // 0 to 16
+    private float texU, texV;
+    // 0 to 0xF0
+    private int overlayU, overlayV;
+    // 0 to 0xF0
+    private int lightU, lightV;
+
+    private final Map<VertexFormatElement, int[]> miscData;
+
+    public VertexData() {
+        this.miscData = new HashMap<>();
+    }
+
+    private VertexData(double posX, double posY, double posZ, float normalX, float normalY, float normalZ, int red, int green, int blue, int alpha, float texU, float texV,
```

Comment:
POJO elements of this class should be done using lombok, for instance `@AllArgsConstructor`